Officer data: Update officers.json and tiers.json

EthanThatOneKid opened the pull request:
In `officers.json`, the order that the officers should be listed is determined by a separate `Tier` (stored in `tiers.json`) which is referenced by ID.

The `officers.json` has some rules that should always be guaranteed:
- Officer position tiers should be synced with the appropriate tier ID.
- Officers are listed in alphabetical order in `officers.json`.

Example updated `Tier`s:
<https://github.com/EthanThatOneKid/acmcsuf.com/blob/414c3f75e97192887231ce9aff505eb8a8b2a940/src/lib/public/board/data/tiers.json#L3-L4>

Example updated `Officer` (using IDs of example tiers above):
<https://github.com/EthanThatOneKid/acmcsuf.com/blob/414c3f75e97192887231ce9aff505eb8a8b2a940/src/lib/public/board/data/officers.json#L63-L68>

EthanThatOneKid committed:
Update officers.json according to new tiers.json

EthanThatOneKid committed:
Resolved lint errors

k4rni commented on src/lib/public/board/data/officers.json:
```diff
@@ -1,718 +1,368 @@
-    },
-    "picture": "jason-anthony.webp"
+    "picture": "jason-anthony.webp",
+    "positions": { "S21": { "title": "Secretary", "tier": 3 } }
```
The updated "tier" for Secretary is '4'

`"Secretary": { "id": 4, "index": 300 },`

EthanThatOneKid replied:
Resolved in <https://github.com/EthanThatOneKid/acmcsuf.com/pull/620/commits/ca60f85521a24ec2b643a1e305918a99674a5c88>.

EthanThatOneKid replied:
I noticed that several review comments were left on this PR that are similar. Consider the above <https://github.com/EthanThatOneKid/acmcsuf.com/pull/620#discussion_r996553848> a reply to your review comments between <https://github.com/EthanThatOneKid/acmcsuf.com/pull/620#discussion_r996526569> and <https://github.com/EthanThatOneKid/acmcsuf.com/pull/620#discussion_r996533968> (inclusive).

k4rni commented on src/lib/public/board/data/officers.json:
```diff
@@ -1,718 +1,368 @@
+      "S21": { "title": "Dev Officer", "tier": 19 },
+      "F21": { "title": "Dev Director", "tier": 17 },
+      "S22": { "title": "Dev President", "tier": 17 },
+      "F22": { "title": "Marketing Co-Lead", "tier": 8 }
```
The updated "tier" for Marketing Leader is '9'

`"Marketing Leader": { "id": 9, "index": 550 }`

k4rni commented on src/lib/public/board/data/officers.json:
```diff
@@ -1,718 +1,368 @@
-    },
-    "picture": "nurhaliza-hassan.webp"
+      "S22": { "title": "Marketing Director", "tier": 8 },
+      "F22": { "title": "Marketing Co-Lead", "tier": 8 }
```
The updated "tier" for Marketing Leader is '9' (both lines 93 and 94)

`"Marketing Leader": { "id": 9, "index": 550 }`

k4rni commented on src/lib/public/board/data/officers.json:
```diff
@@ -1,718 +1,368 @@
-    },
-    "picture": "ibrahim-israr.webp"
+    "picture": "ibrahim-israr.webp",
+    "positions": { "S22": { "title": "Secretary", "tier": 3 } }
```
The updated "tier" for Secretary is '4'

` "Secretary": { "id": 4, "index": 300 },`

k4rni commented on src/lib/public/board/data/officers.json:
```diff
@@ -1,718 +1,368 @@
-    },
-    "picture": "andrew-lau.webp"
+    "picture": "andrew-lau.webp",
+    "positions": { "S21": { "title": "Treasurer", "tier": 4 } }
```
The updated "tier" for Treasurer is '3'

` "Treasurer": { "id": 3, "index": 250 },`

k4rni commented on src/lib/public/board/data/officers.json:
```diff
@@ -1,718 +1,368 @@
-    },
-    "picture": "tommy-le.webp"
+      "F21": { "title": "Treasurer", "tier": 4 },
+      "S22": { "title": "Treasurer", "tier": 4 }
```
The updated "tier" for Treasurer is '3' (both lines 149 and 150)

` "Treasurer": { "id": 3, "index": 250 },`

k4rni commented on src/lib/public/board/data/officers.json:
```diff
@@ -1,718 +1,368 @@
-    },
-    "picture": "nolan-lee.webp"
+    "picture": "nolan-lee.webp",
+    "positions": { "S22": { "title": "Historian", "tier": 9 } }
```
The updated "tier" for Historian is '10'

`"Historian": { "id": 10, "index": 600 },`

jaasonw replied:
false

k4rni commented on src/lib/public/board/data/officers.json:
```diff
@@ -1,718 +1,368 @@
-    },
-    "picture": "jacob-nguyen.webp"
+      "S21": { "title": "President, Create Director, nodebuds Officer", "tier": 0 },
+      "F22": { "title": "Treasurer", "tier": 4 }
```
The updated "tier" for Treasurer is '3' 

`"Treasurer": { "id": 3, "index": 250 }`

k4rni commented on src/lib/public/board/data/officers.json:
```diff
@@ -1,718 +1,368 @@
-    },
-    "picture": "stephanie-pocci.webp"
+      "S22": { "title": "Event Coordinator", "tier": 10 },
+      "F22": { "title": "Event Coordinator", "tier": 10 }
```
The updated "tier" for Event Coordinator is '8' (both lines 252 and 253)

`"Event Coordinator": { "id": 8, "index": 500 },`

k4rni commented on src/lib/public/board/data/officers.json:
```diff
@@ -1,718 +1,368 @@
-    },
-    "picture": "nicolas-renteria.webp"
+    "picture": "nicolas-renteria.webp",
+    "positions": { "S21": { "title": "Marketing Chair", "tier": 8 } }
```
The updated "tier" for Marketing Lead is '9'

`"Marketing Leader": { "id": 9, "index": 550 },`

k4rni commented on src/lib/public/board/data/officers.json:
```diff
@@ -1,718 +1,368 @@
-    },
-    "picture": "wilbert-rodriguez.webp"
+    "picture": "wilbert-rodriguez.webp",
+    "positions": { "S21": { "title": "Intern Program Manager", "tier": 22 } }
```
I suggest to actually switch the tier from "22" to "23". 

`"Nodebuds Officer": { "id": 22, "index": 1200 },`
`"ICC Representative": { "id": 23, "index": 1250 }`

k4rni commented on src/lib/public/board/data/officers.json:
```diff
@@ -1,718 +1,368 @@
-    },
-    "picture": "justin-stitt.webp"
+      "S22": { "title": "Algo Officer", "tier": 12 },
+      "F22": { "title": "AI Team Lead", "tier": 11 }
```
The updated "tier" for AI Lead is '20'

`"AI Leader": { "id": 20, "index": 1100 },`

k4rni commented on src/lib/public/board/data/officers.json:
```diff
@@ -1,718 +1,368 @@
-      }
-    },
-    "picture": "samuel-valls.webp"
+      "S21": { "title": "Community Manager", "tier": 22 },
```
I suggest to actually switch the tier from "22" to "23".

"Nodebuds Officer": { "id": 22, "index": 1200 },
"ICC Representative": { "id": 23, "index": 1250 }

k4rni commented on src/lib/public/board/data/officers.json:
```diff
@@ -1,718 +1,368 @@
-    },
-    "picture": "jason-wong.webp"
+      "S22": { "title": "Historian", "tier": 9 },
+      "F22": { "title": "Historian", "tier": 9 }
```
The updated "tier" for Historian is '10'

`"Historian": { "id": 10, "index": 600 },`

k4rni reviewed (changes requested):
First of all, thank you so much for updating the tiers. This is easy to follow and very clear to understand. There were a couple things that I would like to be clarified:

There were two members who had previously worked as a "workshop manager". Should we include that into the tier list? Perhaps as `"Workshop Manager": { "id": 23, "index": 1250" }` so we can replace `"ICC Representative": { "id": 24", "index": 1300 }`. This is simply a suggestion and is not required!


For consistency, I suggest we update the "nodebuds officer" title to "Nodebuds Officer" to reflect the other titles such as "Dev Officer" or "Algo Officer"

EthanThatOneKid committed:
Attempt to resolve review comments between https://github.com/EthanThatOneKid/acmcsuf.com/pull/620#discussion_r996526569 and https://github.com/EthanThatOneKid/acmcsuf.com/pull/620#discussion_r996533968 (inclusive)

EthanThatOneKid committed:
Capitalized 'Nodebuds' in position titles

Related: https://github.com/EthanThatOneKid/acmcsuf.com/pull/620#pullrequestreview-1143391652.

k4rni reviewed (approved):
Looks great!

EthanThatOneKid committed:
Merge branch 'main' into fix/618
